### ISSUE TITLE
Some images are not showing.

### ISSUE THREAD

**Issue by kkotkkio** [DylanVann/react-native-fast-image#242]
First, Below is showing well.

```
<FastImage
              style={s.image}
              source={{
                uri: 'https://unsplash.it/400/400?image=1',
                headers:{ Authorization: 'someAuthToken' },
                priority: FastImage.priority.normal,
              }}
              resizeMode={FastImage.resizeMode.stretch}
            />
```

But, Below is not showing.

```
<FastImage
              style={s.image}
              source={{
                uri: 'https://s3.ap-northeast-2.amazonaws.com/zzim-app-images/intro/180529_1.png',
                headers:{ Authorization: 'someAuthToken' },
                priority: FastImage.priority.high,
              }}
              resizeMode={FastImage.resizeMode.stretch}
            />
```

Why can't I see any image? 
Can't I load image from s3?

So I test to upload first image in s3. 
1. Download : "https://unsplash.it/400/400?image=1"
2. Uri change : : "https://s3.ap-northeast-2.amazonaws.com/zzim-app-images/test.jpeg)"

But it's not showing too.

**Comment by kkotkkio** [comment]
And I try to use `<Image>` of react-native. It's works. 
The problem is from to use FastImage and S3 Image.

**Comment by kkotkkio** [comment]
I tried this.
```
<FastImage
              style={s.image}
              source={{
                uri: 'https://s3.ap-northeast-2.amazonaws.com/zzim-app-images/intro/180529_1.png',
                headers:{
                  Authorization: 'someAuthToken',
                  'Content-Type': 'image/png'
                },
                priority: FastImage.priority.normal,
              }}
              resizeMode={FastImage.resizeMode.stretch}
            />
```

I added 'Content-Type' but it's still not showing..

**Comment by kkotkkio** [comment]
package.json

```
{
  "name": "ZzimIos",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "@expo/vector-icons": "^6.3.1",
    "native-base": "^2.7.0",
    "prop-types": "^15.6.1",
    "react": "16.4.1",
    "react-native": "^0.56.0",
    "react-native-app-intro-slider": "^0.2.4",
    "react-native-awesome-alerts": "^1.0.6",
    "react-native-blur": "^3.2.2",
    "react-native-carousel-view": "^0.5.1",
    "react-native-fast-image": "^4.0.2",
    "react-native-gifted-chat": "^0.3.0",
    "react-native-keyboard-aware-scroll-view": "^0.4.4",
    "react-native-masonry": "^0.5.0-alpha.1",
    "react-native-navigation": "^2.0.2397",
    "react-native-scalable-image": "^0.2.2",
    "react-native-splash-screen": "^3.0.6",
    "react-native-status-bar-height": "^2.0.0",
    "react-redux": "^5.0.7",
    "recompose": "^0.26.0",
    "redux": "^3.7.2",
    "redux-devtools-extension": "^2.13.2",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.2.0",
    "styled-components": "^3.2.5"
  },
  "devDependencies": {
    "babel-eslint": "^8.0.2",
    "babel-jest": "22.4.3",
    "babel-plugin-module-resolver": "^3.0.0",
    "eslint": "^4.11.0",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.8.0",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-react": "^7.4.0",
    "eslint-plugin-react-native": "^3.2.1",
    "jest": "22.4.3",
    "react-native-debugger-open": "^0.3.17",
    "react-test-renderer": "16.3.1"
  },
  "jest": {
    "preset": "react-native"
  }
}

```

**Comment by jedashford** [comment]
I think this project is dead, no response to any thread.

**Comment by n1ru4l** [comment]
@kkotkkio https://github.com/DylanVann/react-native-fast-image/issues/219

@jedashford just because the maintainer has other stuff to do than responding to the third duplicate of an issue that does not mean the project is dead...

**Comment by kkotkkio** [comment]
@n1ru4l Thank u for answer :) 

I'll still wait for solution to resolve this issue. 
I wrote this issue in StackOverFlow too. 

I wish to use FastImage.

**Comment by dverma4u** [comment]
anyone got solution?

**Comment by n1ru4l** [comment]
@dverma4u did you check #219?

**Comment by dverma4u** [comment]
@n1ru4l  yes i did but my image doesn't have any user agent check. Still not able to display s3 bucket images, but the example image path given in readme works fine

**Comment by QifanViseo** [comment]
I've been read all the solutions above, none of these working. Did anyone got solution? Amazon S3 Images are not working at all.

**Comment by n1ru4l** [comment]
@dverma4u is your s3 image public?

**Comment by IronTony** [comment]
Any news? I am in the same situation and I am using RN 0.58 and latest package version.
I can see the developer image of the example but any JPG/PNG from S3 or google images.

**Comment by IronTony** [comment]
Hint: If you convert the S3 image to base64 and pass to FastImage it seems to work

**Comment by IronTony** [comment]
I confirm. If you convert the image to base64 and pass it to FastImage everything works fine.

**Comment by douglaswissett** [comment]
@IronTony Have you tried testing on a real device? I'm noticing some fast images aren't displaying on my emulator but work fine when on a real device.

**Comment by IronTony** [comment]
@douglaswissett nope. I will do as soon as possible. I will tell you if it will work. Thx

**Comment by IronTony** [comment]
@danhodos sorry for the late. Yep same problem also on real device, both Android and iOS.

**Comment by DucNguyenVan** [comment]
Hi, I just upgraded version to lastest (5.2.0) and it work well.

**Comment by evanjmg** [comment]
I've had this issue as well with 5.1.1, @DucNguyenVan  did upgrading fix it for you?

**Comment by jmiguel77** [comment]
i am trying to retrieve an image from an S3 bucket gateway, that is not public, so it has an x-api-key, but it's not working

```
<FastImage
            style={{ width: '50%', height: 100 }}
            source={{
              uri: item.uri,
              headers: {
                'Content-Type': 'image/jpeg',
                'x-api-key': item.token
              },
              priority: FastImage.priority.high,
              cache: FastImage.cacheControl.web
            }}
            resizeMode={FastImage.resizeMode.stretch}
        />
```

**Comment by evanjmg** [comment]
I get images not showing sometimes as well - it doesn't make a difference for the provider. However, my authorization header works fine for my images. It's usually just cache misses for me

**Comment by rochapablo** [comment]
When the image it's [too big](https://assets.naointendo.com.br/uploads/post/media/26813/content_c0c72fe1-b70e-4b85-885f-922c5993a2a2.jpg), it does not show up on Android.

```
"react": "16.8.3",
"react-native": "0.59.9",
"react-native-fast-image": "^6.0.3",
```

**Comment by ilibilibom** [comment]
I'm also having an issue with Firebase storage images not appearing ... 
```
 <FastImage
            style={[styles.groupActionIcon]}
            source={{uri: groupImage, priority: FastImage.priority.normal}}
            resizeMode={FastImage.resizeMode.contain}
          />
```

**Comment by 0x01001** [comment]
add android:usesCleartextTraffic="true" in AndroidManifest.xml work for me.
	```
  <application
        ...
        android:usesCleartextTraffic="true"
        ...>
        ...
    </application>
```

**Comment by ilibilibom** [comment]
@0x01001 what do you mean by checking date and time ?

**Comment by 0x01001** [comment]
> @0x01001 what do you mean by checking date and time ?

a cause of this problem can be wrong date time of the device, it's about SSL validation failed. 
when I change update automatic time zone it's work.

my log:
```
12-16 22:14:50.118  5821  5821 W Glide   : class com.bumptech.glide.load.engine.GlideException: Failed to load resource
12-16 22:14:50.118  5821  5821 W Glide   : There was 1 cause:
12-16 22:14:50.118  5821  5821 W Glide   : javax.net.ssl.SSLHandshakeException(Chain validation failed)
12-16 22:14:50.118  5821  5821 W Glide   :  call GlideException#logRootCauses(String) for more detail
12-16 22:14:50.118  5821  5821 W Glide   :   Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, REMOTE
12-16 22:14:50.118  5821  5821 W Glide   : There was 1 cause:
12-16 22:14:50.118  5821  5821 W Glide   : javax.net.ssl.SSLHandshakeException(Chain validation failed)
12-16 22:14:50.118  5821  5821 W Glide   :  call GlideException#logRootCauses(String) for more detail
12-16 22:14:50.118  5821  5821 W Glide   :     Cause (1 of 1): class javax.net.ssl.SSLHandshakeException: Chain validation failed
12-16 22:14:50.121  5821  5821 I Glide   : Root cause (1 of 1)
12-16 22:14:50.121  5821  5821 I Glide   : javax.net.ssl.SSLHandshakeException: Chain validation failed
```

**Comment by venix** [comment]
I had the exact same issue with React-Native 0.60+ 

I was trying to load images from S3 bucket url (which were public) 
It seems that resizing the image with the following code, made it work: 

`resizeMode={FastImage.resizeMode.contain}`

Here is the code used, before:
`                  <FastImage
                    style={styles.image}
                    source={{
                      uri: item.src,
                    }}
                  />`
and after: 

`                  <FastImage
                    style={styles.image}
                    source={{
                      uri: item.src,
                    }}
                    resizeMode={FastImage.resizeMode.contain}
                  />`

**Comment by doug-stritt** [comment]
I had issues with some images not showing. Changing the cache mode to web fixed it for me.

`<FastImage
            style={{
              width: imageSize,
              height: imageSize,
              borderRadius: borderRadius,
            }}
            source={{
              uri: imageUrl,
              cache: FastImage.cacheControl.web
            }} 
/>`

**Comment by andreialecu** [comment]
I found out why this was happening, and it only seems to affect iOS, because Android was fine.

Apparently when a source like `{ uri: "https://some.domain.com/image.png" }` is used, the headers being sent out in the download request do not include `Host: some.domain.com`, which will trip most servers. Previously requests were failing with `onError` being called, but without additional information.  (#200 would be nice to have too)

I've had to set a breakpoint in the objective C code to find this and noticed a 404 error being returned by the server.

Manually supplying the host header, like `{ uri: "https://some.domain.com/image.png", headers: { Host: "some.domain.com" } }` makes it work. 

Out project already had fast-image implemented and working, but because of a side effect of AWS S3 signatures, we were issuing headers which included the `host` header in a different part of the app, and that part was working properly. It was only after attempting to use non-aws resources that it started occuring.

Related: #219

cc @DylanVann

**Comment by andreialecu** [comment]
Additionally, I noticed another issue with 302 redirects. When using Google Maps places images, the uri returned for the image is a 302 redirect from `maps.googleapis.com/....` to `lh3.googleusercontent.com/...`.

In this scenario there would need to be two requests, with different `Host` headers. I'm not sure how to workaround it from user-land without fixing the library itself.

Some discussion here: https://github.com/SDWebImage/SDWebImage/issues/2841

**Comment by andreialecu** [comment]
There is a serious problem with http headers, I have opened #690 

Issue was confirmed at https://github.com/SDWebImage/SDWebImage/issues/3031#issuecomment-639376964

**Comment by alessiocancian** [comment]
I'm writing this here because when searching for a solution I found this issue, so this could help someone...

I was experiencing a similar problem, some images were not showing on iOS only...
I found out that those images (even though extension was jpg) were in WEBP format, and SDWebImage doesn't support WEPB for default, but can be added using a pod (#476)

**Comment by jaroslav009** [comment]
same issue

**Comment by anatooly** [comment]
As for me thss problem reproduce on last build 8.5.11, iOS emulator, many images empty(

**Comment by denisk20** [comment]
For my case some images were http, not https, and were not shown. This fixed it:
```
<FastImage
   source={{uri: myUri.replace('http:', 'https:)}} />
```

**Comment by banghia112** [comment]
still experiencing this in ^7.0.2, any suggestions?

**Comment by pratheekgv19** [comment]
https://stackoverflow.com/questions/51296189/some-images-are-not-showing-in-react-native-fastimage
this worked for me
